File: ueueo-framework/ueueo-domain-driven-desgin/src/main/java/com/ueueo/ddd/application/contracts/dtos/EntityDto.cs
using System;

namespace Volo.Abp.Application.Dtos;

[Serializable]
public abstract class EntityDto : IEntityDto //TODO: Consider to delete this class
{
    @Override public string toString()
    {
        return $"[DTO: {GetType().Name}]";
    }
}

[Serializable]
public abstract class EntityDto<TKey> : EntityDto, IEntityDto<TKey>
{
    /// <summary>
    /// Id of the entity.
    /// </summary>
    public TKey Id;// { get; set; }

    @Override public string toString()
    {
        return $"[DTO: {GetType().Name}] Id = {Id}";
    }
}

File: ueueo-frameworks/ueueo-ddd-application/src/main/java/com/ueueo/ddd/application/contracts/dtos/EntityDto.java
package com.ueueo.ddd.application.contracts.dtos;

import com.ueueo.ID;
import lombok.Data;

@Data
public abstract class EntityDto implements IEntityDto {
    /**
     * Id of the entity.
     */
    private ID id;

    @Override
    public String toString() {
        return String.format("[DTO: %s] Id = %s", getClass().getName(), id);
    }
}
